app/src/main/java/com/example/mvvmcodebase/data/source/CocktailRemoteDataSource.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.mvvmcodebase.data.source
+
+import com.example.mvvmcodebase.data.ApiService
+import com.example.mvvmcodebase.data.CocktailDataSource
+import com.example.mvvmcodebase.data.GetDataCallback
+import com.example.mvvmcodebase.data.model.Cocktail
+import com.example.mvvmcodebase.data.model.CocktailResponse
+import com.example.mvvmcodebase.utils.ExceptionConstant
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class CocktailRemoteDataSource(private val apiService: ApiService) : CocktailDataSource {
+
+    private var response: Response<CocktailResponse>? = null
+
+    override suspend fun getCocktails(letter: String, callback: GetDataCallback<List<Cocktail>>) {
+        response = apiService.getDrinks(letter)
+
+        callback.apply {
+            response?.let {
+                if (it.isSuccessful) onSuccess(it.body()?.drinks) else onFailed(it.message())
+            }
+        }
+
+    }
```

Comment:
Ngoài những lỗi do server trả về còn nhiều lỗi khác 
(vd: lỗi convert từ json -> object java )
Để bắt được lỗi này thì em nên try/catch ở ViewModel thay vì bắt lỗi thủ công ở đây

Reply:
Ở đây anh gợi ý cho em cách làm đơn giản nhất để làm quen với Retrofit, coroutine 
```suggestion
    override suspend fun getCocktails(letter: String) =
         apiService.getDrinks(letter).drinks
```